sql/accounts/accounts.go

```diff
@@ -169,3 +170,35 @@ func Revert(db sql.Executor, after types.LayerID) error {
 	}
 	return nil
 }
+
+func IterateAccountsOps(
+	db sql.Executor,
+	operations builder.Operations,
+	fn func(account *types.Account) bool,
+) error {
+	_, err := db.Exec(`select address, balance, next_nonce, max(layer_updated), template, state from accounts`+
+		builder.FilterFrom(operations),
+		builder.BindingsFrom(operations),
+		func(stmt *sql.Statement) bool {
+			var account types.Account
+			stmt.ColumnBytes(0, account.Address[:])
+			account.Balance = uint64(stmt.ColumnInt64(1))
+			account.NextNonce = uint64(stmt.ColumnInt64(2))
+			account.Layer = types.LayerID(uint32(stmt.ColumnInt64(3)))
+			if stmt.ColumnLen(4) > 0 {
+				var template types.Address
+				stmt.ColumnBytes(4, template[:])
+				account.TemplateAddress = &template
+				account.State = make([]byte, stmt.ColumnLen(5))
+				stmt.ColumnBytes(5, account.State)
+			}
+			fn(&account)
+			return true
```

Comment:
```suggestion
			return fn(&account)
```

How about adding a UT to see if iteration is stopped when `fn` returns false?